docs/advanced-guides-nlu.md

```diff
@@ -0,0 +1,204 @@
+
+`.env`
+
+```
+RESOURCE_NAME=
+KNOWLEDGE_BASE_ID=
+ENDPOINT_KEY=
+```
+
+Next, we can get the answer by calling the API endpoint. In the example, we use `axios` as our HTTP client. For other Node HTTP clients like `request-promise`, you can see this [document](https://docs.microsoft.com/en-us/azure/cognitive-services/qnamaker/quickstarts/get-answer-from-knowledge-base-nodejs). The returned response will contain an array with possible QA pairs and corresponding scores. You can define your own score threshold in the code.
+
+```js
+const axios = require('axios');
+
+const { RESOURCE_NAME, KNOWLEDGE_BASE_ID, ENDPOINT_KEY } = process.env;
+
+module.exports = async function App(context) {
+  if (context.event.isText) {
+    const { data } = await axios.post(
+      `https://${RESOURCE_NAME}.azurewebsites.net/qnamaker/knowledgebases/${KNOWLEDGE_BASE_ID}/generateAnswer`,
+      { question: context.event.text },
+      {
+        headers: {
+          Authorization: `EndpointKey ${ENDPOINT_KEY}`,
+        },
+      }
+    );
+
+    const topAnswer = data.answers[0];
+
+    // You can define your own score threshold here.
+    if (topAnswer.score > 70) {
+      await context.sendText(topAnswer.answer);
+    }
+  }
+};
+```
+
+You can check out the full example [Here](https://github.com/Yoctol/bottender/tree/master/examples/with-qna-maker).
+
+## Building with LUIS
+
+To build a bot integrated with [LUIS (Language Understanding Intelligent Service)](https://luis.ai/), you have to create a new app in the LUIS portal following the [Official Setup Guide](https://docs.microsoft.com/en-us/azure/cognitive-services/LUIS/get-started-portal-build-app) and fill the three values: `LUIS_APP_ID`, `LUIS_APP_KEY`, and `LUIS_APP_ENDPOINT` into the `.env` file.
+
+`.env`
+
+```
+LUIS_APP_ID=
+LUIS_APP_KEY=
+LUIS_APP_ENDPOINT=
+```
+
+Next, you can add a simple intent. In this example, we create a simple intent with the intent name `greeting`. You can set your own training phrases on the LUIS console for this intent. And then you have to train the model and publish it.
+
+After publishing successfully, you can call LUIS's API to detect the intention of the message the bot receives. In this example, we use `axios` as our HTTP client to call the API.
+
+```js
+const querystring = require('querystring');
+const axios = require('axios');
+
+const { LUIS_APP_KEY, LUIS_APP_ENDPOINT, LUIS_APP_ID } = process.env;
+
+module.exports = async function App(context) {
+  if (context.event.isText) {
+    const queryParams = {
+      verbose: true,
+      q: context.event.text,
+      'subscription-key': LUIS_APP_KEY,
+    };
+
+    const { data } = await axios.get(
+      `${LUIS_APP_ENDPOINT}luis/v2.0/apps/${LUIS_APP_ID}?${querystring.stringify(
+        queryParams
```

Comment:
Hey @HcwXd, you can use `params` to simplify those `querystring.stringify` stuff here.

Reply:
Thanks for pointing this out! I forget to fix this when revising the code in the #544